e2e/cypress/integration/bot_accounts/bot_api_spec.js

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @bot_accounts
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Bot accounts ownership and API', () => {
+    let newTeam;
+    let botName;
+    let newUser;
+    let newChannel;
+    let newBot;
+    let adminUser;
+    beforeEach(() => {
+        cy.apiAdminLogin().then(({user}) => {
+            adminUser = user;
+        });
+
+        cy.apiInitSetup().then(({team, user, channel}) => {
+            newTeam = team;
+            newUser = user;
+            newChannel = channel;
+        });
+
+        botName = 'bot-' + Date.now();
+
+        // # Set ServiceSettings to expected values
+        const newSettings = {
+            ServiceSettings: {
+                EnforceMultifactorAuthentication: false,
+                EnableBotAccountCreation: true,
+                DisableBotsWhenOwnerIsDeactivated: true,
+            },
+            PluginSettings: {
+                Enable: true,
+                RequirePluginSignature: false,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Create a test bot
+        cy.apiCreateBot(botName, 'Test Bot', 'test bot').then(({bot}) => {
+            newBot = bot;
+            cy.apiPatchUserRoles(bot.user_id, ['system_admin', 'system_user']);
+        });
+    });
+
+    it('MM-T1872 Bot can post to DM channel', () => {
+        // # Login as admin
+        cy.apiAdminLogin();
```

Comment:
This seems redundant since user session didn't change since `beforeEach`. Same for other `it` blocks.